extensions/ql-vscode/src/vscode-tests/no-workspace/remote-queries/variant-analysis-manager.test.ts

```diff
@@ -44,9 +44,21 @@ describe('Variant Analysis Manager', async function() {
     variantAnalysis = createMockApiResponse('in_progress', scannedRepos);
 
     try {
-      const extension = await extensions.getExtension<CodeQLExtensionInterface | Record<string, never>>('GitHub.vscode-codeql')!.activate();
-      cli = extension.cliServer;
-      variantAnalysisManager = new VariantAnalysisManager(extension.ctx, cli, storagePath, logger);
+      const ctx = createMockExtensionContext();
+      cli = createMockCliServer({
+        bqrsInfo: [{ 'result-sets': [{ name: 'result-set-1' }, { name: 'result-set-2' }] }],
+        bqrsDecode: [{
+          columns: [{ kind: 'NotString' }, { kind: 'String' }],
+          tuples: [['a', 'b'], ['c', 'd']],
+          next: 1
+        }, {
+          columns: [{ kind: 'String' }, { kind: 'NotString' }, { kind: 'StillNotString' }],
+          tuples: [['a', 'b', 'c']]
+        }]
```

Comment:
It seems like these mock operations are not actually used in these tests. Would it make sense to switch the order of the arguments and make the `mockOperations` optional, so that this can be simplified to `cli = createMockCliServer(sandbox)`?

Reply:
I've changed this to make the `mockOperations` argument optional: https://github.com/github/vscode-codeql/pull/1628/commits/4341354c45447f0a9996f9bdaf151e4287c61ab8 & https://github.com/github/vscode-codeql/pull/1628/commits/91523bdadc8423f63b4382bd01f3a574ac1eb630